src/resmom/catch_child.c

```diff
@@ -326,72 +333,66 @@ static enum job_atr mom_rtn_list_ext[] = {
  * @retval	NULL		 -	if not successful, filling out 'msg'
  *					with the actual error message.
  */
-PyObject *
+static PyObject *
 json_loads(char *value, char *msg, size_t msg_len)
 {
-	PyObject	*py_name = NULL;
-	PyObject	*py_module = NULL;
-	PyObject 	*py_dict = NULL;
-	PyObject	*py_func_loads = NULL;
-	PyObject	*py_value = NULL;
-	PyObject	*py_result = NULL;
+	PyObject *py_value = NULL;
+	PyObject *py_result = NULL;
 
-	if (value == NULL) {
+	if (value == NULL)
 		return NULL;
-	}
 
 	if (msg != NULL) {
-		if (msg_len <= 0) {
+		if (msg_len <= 0)
 			return NULL;
-		}
 		msg[0] = '\0';
 	}
 
-	py_name = PyUnicode_FromString("json");
-	if (py_name == NULL) {
-		if (msg != NULL) {
-			snprintf(msg, msg_len, "failed to construct json name");
+	if (py_json_name == NULL) {
+		py_json_name = PyUnicode_FromString("json");
+		if (py_json_name == NULL) {
+			if (msg != NULL)
+				snprintf(msg, msg_len, "failed to construct json name");
+			return NULL;
```

Comment:
we can put goto here also, but it not really needed because at this point py_value and py_result always be NULL.
If u reach this point means, json_loads/dumps are called very first time in the process, and now it trying to import json module which failed, so there is nothing to free here.

But you give me good point, we don't need below goto(s) also as we are not freeing global static variable (aka `py_json_name`, `py_json_module`, `py_json_dict` and `py_json_func_loads`) so no need of goto till we have `py_value` != NULL. will change it.